lib/ews/connection.rb

```diff
@@ -80,13 +83,22 @@ def hostname
   # @param opts [Hash] misc opts for handling the Response
   def dispatch(ews, soapmsg, opts)
     respmsg = post(soapmsg, options: opts)
+
+    respmsg, resp_headers = respmsg if respmsg.is_a?(Array)
```

Comment:
probably use `respbody` instead of reuse `resmsg` so we still have the original data?